web3_proxy/src/frontend/users/payment.rs

```diff
@@ -413,60 +421,26 @@ pub async fn user_balance_post(
         // Check if the item is in the database. If it is not, then add it into the database
         let user_balance = balance::Entity::find()
             .filter(balance::Column::UserId.eq(recipient.id))
-            .one(&db_conn)
-            .await?;
+            .lock(LockType::Update)
+            .one(&txn)
+            .await?
+            .context("Could not find User balance, this should have been created when signing up the user!")?;
```

Comment:
For new users, sure. But old users won't have a row. Just default to 0

Reply:
I was thinking that we should backfill all balances to 0 when applying the migrations. I think it would easier as we remove edge-cases and keep the data more uniform. What do you think?